lib/ex_uid2/api.ex

```diff
@@ -0,0 +1,30 @@
+defmodule ExUid2.Api do
+  alias ExUid2.Api.Request
+  alias ExUid2.Api.Response
+
+  def post_encrypted_request(path, payload, ts \\ :os.system_time(:millisecond)) do
+    secret_key = get_secret_key()
+
+    with {:request_body, {:ok, request_body}} <- {:request_body, Request.make_encrypted_request_body(payload, secret_key, ts)},
+         {:response, {:ok, %Req.Response{status: 200, body: response_body}}} <- {:response, post(request_body, path)},
+         {:parsed_response, {:ok, parsed_response}} <- {:parsed_response, Response.decrypt_and_parse(response_body, secret_key)} do
+      {:ok, parsed_response}
+    else
+      {:request_body, error} -> error
+      {:response,{:ok, %Req.Response{status: _}} = response} -> {:error, response}
+      {:response, error} -> error
+      {:parsed_response, error} -> error
```

Comment:
If you match `{:response, {:ok, %Req.Response{status: _}} = response}` as the first head here, you could combine the rest of them into `{_, error} -> error`. I'm not sure if that's a good idea or not, but it's a possibility.

Reply:
Oh indeed, we're not doing anything with those atoms anyway.